packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useManagedVcdDatacentres.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import { useQuery } from '@tanstack/react-query';
+import {
+  getVcdDatacentre,
+  getVcdDatacentreQueryKey,
+  getVcdDatacentres,
+  getVcdDatacentresQueryKey,
+  TGetVcdDatacentreParams,
+} from '../api/hpc-vmware-managed-vcd-datacentre';
+import IVcdDatacentre from '@/types/vcd-datacenter.interface';
+
+const useManagedVcdDatacentres = (id: string) => {
+  return useQuery<ApiResponse<IVcdDatacentre[]>, ApiError>({
+    queryKey: getVcdDatacentresQueryKey({ id }),
+    queryFn: () => getVcdDatacentres({ id }),
+    retry: false,
+    ...{
+      keepPreviousData: true,
+    },
+  });
+};
+
+export const useManagedVcdDatacentre = ({
+  id,
+  vdcId,
+}: TGetVcdDatacentreParams) => {
```

Comment:
**question:** What is the need to accept params as object? it can be strings right?

Reply:
yes, fixed !